crates/forge/bin/cmd/inspect.rs

```diff
@@ -69,12 +72,30 @@ impl InspectArgs {
         } else {
             project.find_contract_path(&contract.name)?
         };
-        let mut output = compiler.files([target_path.clone()]).compile(&project)?;
 
-        // Find the artifact
-        let artifact = output.remove(&target_path, &contract.name).ok_or_else(|| {
-            eyre::eyre!("Could not find artifact `{contract}` in the compiled artifacts")
-        })?;
+        let artifact = if modified_build_args.compiler.zk.enabled() {
+            let config = Config {
+                solc: Some(SolcReq::Version(Version::new(0, 8, 26))),
+                ..Default::default()
+            };
+            let project = config_create_project(&config, false, true).unwrap();
+            compiler
+                .files([target_path.clone()])
+                .zksync_compile(&project)?
+                .remove(&target_path, &contract.name)
+                .ok_or_else(|| {
+                    eyre::eyre!("Could not find artifact `{contract}` in the compiled artifacts")
+                })?
+                .into()
+        } else {
+            compiler
+                .files([target_path.clone()])
+                .compile(&project)?
+                .remove(&target_path, &contract.name)
+                .ok_or_else(|| {
+                    eyre::eyre!("Could not find artifact `{contract}` in the compiled artifacts")
+                })?
+        };
```

Comment:
```suggestion
```suggestion
        let mut output = compiler.files([target_path.clone()]).compile(&project)?;

        // Find the artifact
        let artifact = output.remove(&target_path, &contract.name).ok_or_else(|| {
            eyre::eyre!("Could not find artifact `{contract}` in the compiled artifacts")
        })?;
        artifact
```
Ideally the upstream code is untouched and we add `NOTE(zk)` when we need to modify it

Reply:
Also try to make the zksync code as analogous to the original as possible so the coginitive overload of comparing both is as small as possible

Reply:
I followed this approach; take a look and let me know.